app/src/main/java/org/simple/clinic/bloodsugar/entry/BloodSugarEntrySheet.kt

```diff
@@ -267,8 +267,12 @@ class BloodSugarEntrySheet : BottomSheetActivity(), BloodSugarEntryUi, RemoveBlo
     }
   }
 
-  override fun showBloodSugarLowError() {
-    showBloodSugarErrorMessage(getString(R.string.bloodsugarentry_error_lower_limit))
+  override fun showBloodSugarLowError(measurementType: BloodSugarMeasurementType) {
+    when (measurementType) {
+      Random, PostPrandial, Fasting -> showBloodSugarErrorMessage(getString(R.string.bloodsugarentry_error_lower_limit))
+      HbA1c -> showBloodSugarErrorMessage(getString(R.string.bloodsugarentry_error_lower_limit_hba1c))
+      is Unknown -> throw AssertionError("Unknown measurement type: $measurementType")
```

Comment:
Should we be throwing an error here? Or maybe we do not perform validation for these cases at all.
Let's check what should be expected here.

Reply:
I have removed the assertion error and instead disallowed editing of blood sugars if the type is unknown